renku/core/management/datasets.py

```diff
@@ -207,16 +212,15 @@ def _add_from_url(self, dataset, path, url, link=False, **kwargs):
                 '{} URLs are not supported'.format(u.scheme)
             )
 
-        # Respect the directory struture inside the source path.
-        relative_to = kwargs.pop('relative_to', None)
-        if relative_to:
-            dst_path = Path(u.path).resolve().absolute().relative_to(
-                Path(relative_to).resolve().absolute()
-            )
-        else:
-            dst_path = os.path.basename(u.path)
+        if destination:
+            destination = self._resolve_paths(dataset_path,
+                                              [destination]).pop()
+
+        dst = self.path / dataset_path / destination
 
-        dst = path.joinpath(dst_path).absolute()
+        # FIXME raise if destination is a file and source is a directory
```

Comment:
Can we open a ticket for this and put the ticket number in the comment?

Reply:
Actually it was easier to fix this.